presto-array/src/main/java/com/facebook/presto/array/SliceBigArray.java

```diff
@@ -61,22 +62,32 @@ public void set(long index, Slice value)
     {
         Slice currentValue = array.get(index);
         if (currentValue != null) {
```

Comment:
it would be nice to extract the reference manipulation code to a separate method so `set` method becomes:

```
    public void set(long index, Slice value)
    {
        updateRetainedSize(index, value);
        array.set(index, value);
    }
```